src/components/vacation-requests-table/vacation-requests-table-toolbar/vacation-requests-table-toolbar.tsx

```diff
@@ -0,0 +1,144 @@
+import { Add, Cancel, Edit } from "@mui/icons-material";
+import { Box, Collapse, Grid, Typography, styled } from "@mui/material";
+import { useEffect, useState } from "react";
+import ToolbarForm from "./toolbar-form/toolbar-form";
+import { GridRowId } from "@mui/x-data-grid";
+import { DataGridRow, ToolbarFormModes, VacationData } from "../../../types";
+import ToolbarDeleteButton from "./toolbar-delete-button";
+import FormToggleButton from "./toolbar-form-toggle-button";
+import ConfirmationHandler from "../../contexts/confirmation-handler";
+import strings from "../../../localization/strings";
+import { getToolbarTitle } from "../../../utils/toolbar-utils";
+
+/**
+ * Component properties
+ */
+interface Props {
+  deleteVacationRequests: (selectedRowIds: GridRowId[], rows: DataGridRow[]) => Promise<void>;
+  createVacationRequest: (vacationData: VacationData) => Promise<void>;
+  updateVacationRequest: (vacationData: VacationData, vacationRequestId: string) => Promise<void>;
+  setFormOpen: (formOpen: boolean) => void;
+  formOpen: boolean;
+  selectedRowIds: GridRowId[];
+  rows: DataGridRow[];
+}
+/**
+ * Table toolbar component
+ *
+ * @param props component properties
+ */
+const TableToolbar = ({
+  deleteVacationRequests,
+  createVacationRequest,
+  updateVacationRequest,
+  setFormOpen,
+  formOpen,
+  selectedRowIds,
+  rows
+}: Props) => {
+  const [toolbarOpen, setToolbarOpen] = useState(false);
+  const [toolbarFormMode, setToolbarFormMode] = useState<ToolbarFormModes>(ToolbarFormModes.NONE);
+  const [confirmationHandlerOpen, setConfirmationHandlerOpen] = useState(false);
+  const [title, setTitle] = useState(strings.tableToolbar.myRequests);
+
+  useEffect(() => {
+    if (toolbarFormMode) {
+      setTitle(getToolbarTitle(toolbarFormMode));
+    }
```

Comment:
Generally its good to just call a function in a useEffect rather than having any logic or conditions. Coudl you move the if toolbarFormMode  check into getToolbarTitle. Not a big deal if not.

Reply:
Removed the conditional as the parameter is required in getToolbarTitle